apps/api/migrations/subscribers/remove-duplicated-subscribers/remove-duplicated-subscribers.migration.spec.ts

```diff
@@ -0,0 +1,264 @@
+      data: { newStuff: 'value_4' },
+      _environmentId: firstEnvironmentId,
+      _organizationId: session.organization._id,
+    });
+
+    await subscriberRepository.create({
+      subscriberId: duplicatedSubscriberId,
+      firstName: 'first_name_5',
+      locale: 'locale_5',
+      _environmentId: firstEnvironmentId,
+      _organizationId: session.organization._id,
+    });
+
+    const duplicates = await subscriberRepository.find({
+      subscriberId: duplicatedSubscriberId,
+      _environmentId: session.environment._id,
+    });
+    expect(duplicates.length).to.equal(5);
+
+    await removeDuplicatedSubscribers();
+
+    const remainingDuplicates = await subscriberRepository.find({
+      subscriberId: duplicatedSubscriberId,
+      _environmentId: session.environment._id,
+    });
+    expect(remainingDuplicates.length).to.equal(1);
+    expect(remainingDuplicates[0]._id).to.equal(firstCreatedSubscriber._id);
+    expect(remainingDuplicates[0]._organizationId).to.equal(firstCreatedSubscriber._organizationId);
+    expect(remainingDuplicates[0]._environmentId).to.equal(firstCreatedSubscriber._environmentId);
+    expect(remainingDuplicates[0].__v).to.equal(firstCreatedSubscriber.__v);
+
+    expect(remainingDuplicates[0].firstName).to.equal('first_name_5');
+    expect(remainingDuplicates[0].lastName).to.equal('last_name_1');
+    expect(remainingDuplicates[0].email).to.equal('email_3');
+    expect(remainingDuplicates[0].phone).to.equal('phone_3');
+    expect(remainingDuplicates[0].avatar).to.equal('avatar_4');
+    expect(remainingDuplicates[0].locale).to.equal('locale_5');
+    expect(remainingDuplicates[0].data?.key).to.be.undefined;
+    expect(remainingDuplicates[0].data?.newStuff).to.equal('value_4');
+  });
+
+  it('should keep the first created subscriber after merge', async () => {
+    const duplicatedSubscriberId = '123';
+    const firstEnvironmentId = session.environment._id;
+
+    const firstCreatedSubscriber = await subscriberRepository.create({
+      subscriberId: duplicatedSubscriberId,
+      firstName: 'first_name',
+      _environmentId: firstEnvironmentId,
+      _organizationId: session.organization._id,
+    });
+    await subscriberRepository.create({
+      subscriberId: duplicatedSubscriberId,
+      firstName: 'first_name_2',
+      _environmentId: firstEnvironmentId,
+      _organizationId: session.organization._id,
+    });
+
+    const duplicates = await subscriberRepository.find({
+      subscriberId: duplicatedSubscriberId,
+      _environmentId: session.environment._id,
+    });
+    expect(duplicates.length).to.equal(2);
+
+    await removeDuplicatedSubscribers();
+
+    const remainingDuplicates = await subscriberRepository.find({
+      subscriberId: duplicatedSubscriberId,
+      _environmentId: session.environment._id,
+    });
+
+    expect(remainingDuplicates.length).to.equal(1);
+    expect(remainingDuplicates[0]._id).to.equal(firstCreatedSubscriber._id);
+  });
+});
```

Comment:
✏ **suggestion:** it would be nice to see some tests with the properly merged `channels`

Reply:
@LetItRock Done :)